### PR TITLE
Components to Typescript

### PR DESCRIPTION
**Pull request by IvanFeng1**
* Panel
* LogInteractionEvents
* CountdownCircle

### CODE REVIEW

**Comment by IvanFeng1** [comment]
https://github.com/TBS-EACPD/infobase/blob/a2f36cbd3ac2e87b6b2475cb986862875ed2100a/client/src/core/ensure_loaded.ts#L426-L433

I am currently getting an error on Line 433 because `.uniqBy()` requires a parameter but the javascript code had no parameter for this function. My workaround for this locally is making  parameters not required for `.uniqBy()`  in the function definition, which is in the node_modules but I don't know how to fix this for the remote repo.

**Comment by Stephen-ONeil** [comment]
The default `uniqBy` function is `_.identity` https://lodash.com/docs/4.17.15#uniqBy. That'll be the case for most lodash utils that take a predicate function. A little annoying the lodash types doesn't account for that.

File: client/src/components/LogInteractionEvents.tsx [hunk 1]
```diff
@@ -3,7 +3,14 @@ import React from "react";
 
 import { log_standard_event } from "src/core/analytics";
 
-export const LogInteractionEvents = ({
+interface LogInterationEventsProps {
+  event_type: string;
+  event_details: string;
+  style: { display: string };
```
Comment (ktw1016) [inline client/src/components/LogInteractionEvents.tsx:9]: You can use `React.CSSProperties` here

File: client/src/components/LogInteractionEvents.tsx [hunk 2]
```diff
@@ -3,7 +3,14 @@ import React from "react";
...
+  children: React.ReactElement;
```
Comment (ktw1016) [inline client/src/components/LogInteractionEvents.tsx:10]: ReactNode over ReactElement I think

File: client/src/components/Panel.tsx [hunk 3]
```diff
@@ -14,7 +14,59 @@ import "./Panel.scss";
...
+  is: (comparator: any) => boolean;
+  legal_title: string;
+  level: string;
+  lookup: Function;
+  name: string;
+  plural: string;
+  singular: string;
+  subject_type: string;
+}
+interface Footnote {
```
Comment (ktw1016) [inline client/src/components/Panel.tsx:49]: I would rather you drop converting Panel.js for now for the same reason as ensure_loaded.
These footnotes, subject should be built from the ground up

**Review by ktw1016 — CHANGES_REQUESTED** [review]

**Review by ktw1016 — APPROVED** [review]
👍👍

**Pull request merged**